src/js/tracks/text-track-fieldset.js

```diff
@@ -0,0 +1,129 @@
+import Component from '../component';
+import * as Dom from '../utils/dom';
+import * as Guid from '../utils/guid';
+import TextTrackSelect from './text-track-select';
+
+/**
+ * Creates fieldset section of 'TextTrackSettings'.
+ * Manganes two versions of fieldsets, one for type of 'colors'
+ * & the other for 'font', Component adds diferent DOM elements
+ * to that fieldset  depending on the type.
+ *
+ * @extends Component
+ */
+class TextTrackFieldset extends Component {
+
+  /**
+   * Creates an instance of this class.
+   *
+   * @param { import('./player').default } player
+   *        The `Player` that this class should be attached to.
+   *
+   * @param {Object} [options]
+   *        The key/value store of player options.
+   *
+   * @param { import('../utils/dom').ContentDescriptor} [options.content=undefined]
+   *        Provide customized content for this modal.
+   *
+   * @param {string} [options.legendId]
+   *        A text with part of an string to create atribute of aria-labelledby.
+   *        It passes to 'TextTrackSelect'.
+   *
+   * @param {string} [options.id]
+   *        A text with part of an string to create atribute of aria-labelledby.
+   *        It passes to 'TextTrackSelect'.
+   *
+   * @param {string} [options.legendText]
```

Comment:
I'd double check that the documentation for these new track related components are accurate. For example, I don't think we need `options.legendText` here since we aren't actually ever using legendText passed through that `options` object. I think this is the case for a handful of places in these components.

Reply:
Hello @wseymour15  legend Text is used to create inside the component like this: 

```
const legendElement = Dom.createEl('legend', {
      textContent: this.localize(this.options_.legendText),
      id: this.options_.legendId
});
```

Is also used in every new instance of the component
<img width="406" alt="Screenshot 2024-04-16 at 10 34 51" src="https://github.com/videojs/video.js/assets/21351830/e1f08c28-183b-446b-8fd9-d27717d9e694">

Reply:
Fair enough, I guess it is truly used in a component that this extends, but you are probably right that it is fine to leave here if it is common usage.